solutions/devsprint-1/DeliveryAppChallenge/Service/NetworkManager.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  NetworkManager.swift
+//  DeliveryAppChallenge
+//
+//  Created by Hellen on 02/11/21.
+//
+
+import Foundation
+
+typealias NetworkResult<T: Decodable> = ((Result<T, APIError>) -> Void)
+
+protocol NetworkManagerProtocol {
+    
+    static func request<T: Decodable>(url: String, completion: @escaping NetworkResult<T>)
+}
+
+class NetworkManager: NetworkManagerProtocol {
+    
+    static func request<T>(url: String, completion: @escaping NetworkResult<T>) {
+        guard let url = URL(string: url) else { return }
+        
+        let dataTaks = URLSession.shared.dataTask(with: url) { data, _, _ in
```

Comment:
{ data, _, _ in -> aqui vc poderia verificar a response e erro
Ex: 
 let task = session.dataTask(with: url) { (data, response, error) in
                if error != nil {
                    //Fazer algum tratamento
                    return
                }
                guard let response = response as? HTTPURLResponse, response.statusCode == 200 else {
                   //Fazer algum tratamento
                    return
                }
Acho que o Alamofire costuma fazer isso por trás dos panos, mais com a URL Session talvez seja legal verificar tbm

Reply:
O que acha de injetarmos a URLSession e passar o shared na classe de `NetworkManager`? 
Dessa forma a gente consegue criar um protocolo implementando o método de `dataTask(urlRequest:)` e conseguir testar o comportamento desse método, além de podermos fazer um mock da closure que será feito o trigger, etc.